modules/class-simplechart-plugin-versions.php

```diff
@@ -0,0 +1,132 @@
+	private $_latest_plugin_zip_url;
+	private $_latest_plugin_update_name;
+	private $_latest_plugin_update_description;
+	private $_latest_plugin_last_updated;
+
+	public function __construct() {
+		add_action( 'admin_init', array( $this, 'simplechart_check_for_new_version' ) );
+		add_filter( 'plugins_api_result', array( $this, 'simplechart_mock_plugins_api' ), 10, 3 );
+	}
+
+	public function simplechart_check_for_new_version() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		if ( false === ( $this->_latest_plugin_version = get_transient( 'simplechart_plugin_version_remote' ) ) ) {
+			$this->update_simplechart_remote_version();
+			return;
+		}
+
+		if ( false === ( $this->_latest_plugin_zip_url = get_transient( 'simplechart_plugin_zip_url_remote' ) ) ) {
+			$this->_latest_plugin_zip_url = $this->_simplechart_url;
+		}
+
+		if ( false === ( $this->_latest_plugin_update_name = get_transient( 'simplechart_plugin_update_name' ) ) ) {
+			$this->_latest_plugin_update_name = __( 'Latest Update', 'simplechart' );
+		}
+
+		if ( false === ( $this->_latest_plugin_update_description = get_transient( 'simplechart_plugin_update_body' ) ) ) {
+			$this->_latest_plugin_update_description = __( 'See more on Github.', 'simplechart' );
+		}
+
+		$this->_latest_plugin_last_updated = get_transient( 'simplechart_plugin_update_last_updated' );
+
+		if ( version_compare( Simplechart::instance()->get_config( 'version' ), $this->_latest_plugin_version, '<' ) ) {
+			add_filter( 'site_transient_update_plugins', array( $this, 'simplechart_extend_filter_update_plugins' ) );
+			add_filter( 'transient_update_plugins', array( $this, 'simplechart_extend_filter_update_plugins' ) );
+		}
+	}
+
+	public function simplechart_extend_filter_update_plugins( $update_plugins ) {
+		if ( ! is_object( $update_plugins ) ) {
+			return $update_plugins;
+		}
+
+		if ( ! isset( $update_plugins->response ) || ! is_array( $update_plugins->response ) ) {
+			$update_plugins->response = array();
+		}
+
+		$update_plugins->response['wordpress-simplechart/simplechart.php'] = (object) array(
+			'slug'         => 'wordpress-simplechart',
+			'new_version'  => $this->_latest_plugin_version,
+			'url'          => $this->_simplechart_url,
+		);
+		return $update_plugins;
+	}
+
+	public function simplechart_mock_plugins_api( $res, $action, $args ) {
+		if ( 'plugin-information' !== $action && 'wordpress-simplechart' !== $args->slug ) {
+			return $res;
+		}
+		$copy_generated = sprintf(
+			__( 'Ask an administrator to replace your existing Simplechart plugin directory with <a href="%1$s">the ZIP of %2$s via Github</a>', 'simplechart' ),
+			esc_url( $this->_latest_plugin_zip_url ),
+			$this->_latest_plugin_version
+		);
+		$description =
+			'<small>'
+			. $copy_generated
+			. '</small>'
+			. '<h1>'
+			. $this->_latest_plugin_update_name
+			. '</h1>'
+			. $this->_latest_plugin_update_description;
+
+		$fake_response = (object) array(
```

Comment:
Instead of creating the `$fake_response` object & then returning, you could instead just return this object.